wordpress-mobile/WordPress-Android
# My Site Dashboard - Phase 2 - Draft and Scheduled Post Cards

Parent #15200

This PR adds draft and scheduled post cards using mocked data.

  ![light_mode_post_cards](https://user-images.githubusercontent.com/1405144/141112514-38a7789d-9229-4e73-862d-4ddab23abcb4.png)

To test:

1. Go to `My Site` -> `Me` -> `App Settings` -> `Test Feature Configuration`.
2. Turn the `MySiteDashboardPhase2FeatureConfig` feature flag `ON`.
3. Relaunch the app.
4. Note that on the `My Site` tab
    - Draft and Scheduled post cards are present initially.
    - Card visibility based on draft/ scheduled posts availability is handled (can be tested by updating mocked data).
    - Null title/excerpt/ featured image is handled (can be tested by updating mocked data).
    - RTL is supported.

Notes:
1. The PR includes basic styling and doesn't require a design review yet. There's a separate fine-tune UI task which will include a design review for all post cards. 
2. When no title is present for a post, "Untitled" is shown in the title. 
3. As discussed, single post item with two-line excerpt is not supported due to UI complexity. If needed, we'll get back to it in a future PR. 

## Regression Notes
1. Potential unintended areas of impact
To be consistent with post cards, currently, no text style changes are applied to other cards (e.g.  Quick Links or Quick Start Cards) as post cards are behind a feature flag. Will discuss and apply them in the fine-tune UI task.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
None

3. What automated tests I added (or what prevented me from doing so)
Added unit tests for new post cards.

PR submission checklist:
